lib/stages/basebandReadout.cpp

```diff
@@ -171,149 +185,192 @@ void basebandReadout::main_thread() {
+                             (dump_statuses[freqidx]->request).event_id, freq_ids[freqidx]);
+                        dump_statuses[freqidx]->state = basebandDumpStatus::State::ERROR;
+                        dump_statuses[freqidx]->finished =
+                            std::make_shared<std::chrono::system_clock::time_point>(
+                                std::chrono::system_clock::now());
+                        switch (data.status) {
+                            case basebandDumpData::Status::TooLong:
+                                dump_statuses[freqidx]->reason =
+                                    "Request length exceeds the configured limit.";
+                                break;
+                            case basebandDumpData::Status::Late:
+                                dump_statuses[freqidx]->reason = "No data captured.";
+                                request_no_data_counters[0]->inc();
+                                break;
+                            case basebandDumpData::Status::ReserveFailed:
+                                dump_statuses[freqidx]->reason =
+                                    "No free space in the baseband buffer";
+                                break;
+                            case basebandDumpData::Status::Cancelled:
+                                dump_statuses[freqidx]->reason = "Kotekan exiting.";
+                                break;
+                            default:
+                                INFO("Unknown dump status: {}", int(data.status));
+                                throw std::runtime_error("Unhandled basebandDumpData::Status case "
+                                                         "in a switch statement.");
+                        }
+                    } else {
+                        INFO("Captured {:d} samples for event {:d} and freq {:d}.",
+                             data.data_length_fpga, data.event_id, data.freq_id);
+
+                        // we are done copying the samples into the readout buffer
+                        mgrs[freqidx]->ready(
+                            {*(dump_statuses[freqidx]),
+                             dumps_to_write_vec.back()}); // TODO: Can we use data or do we need
+                                                          // dumps_to_write_vec.back()?
                     }
                 }
-            }
-
-            if (data.status == basebandDumpData::Status::Ok) {
-                INFO("Captured {:d} samples for event {:d} and freq {:d}.", data.data_length_fpga,
-                     data.event_id, data.freq_id);
-
-                // we are done copying the samples into the readout buffer
-                mgr.ready({dump_status, data});
+                milliseconds ms_after =
+                    duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+                // INFO("memcpy() call: {:d} = {:d}-{:d} ms", ms_after.count() - ms_before.count(),
```

Comment:
Because this was removed (I'm not sure why?). This is generating unused variable warnings.  If we want to add it back we should make this DEBUG, and wrap these variables and calls in `#ifdef DEBUGGING`.

Reply:
I don't think we actually want these INFO statements back or the timing variables--this was to quickly test the speed of memcpy() vs nt_memcpy() and I never changed it because things were working and Kiyo mumbled something once about print statements affecting the stability of multi-threaded processes, but that's not a good reason for having it in there.